### PR TITLE
Add video support on the editor

### PR DESCRIPTION
**Pull request by SergioEstevao**
Add support for video insertion on the visual editor.

cc @bummytime

### CODE REVIEW

File: WordPress/Classes/Models/Media.m [hunk 1]
```diff
@@ -530,4 +530,12 @@ - (NSString *)html
     return result;
 }
 
+- (NSString *)thumbnaiLocalURL;
```
Comment (bummytime) [inline WordPress/Classes/Models/Media.m]: Typo — `thumbnailLocalURL` ?

**Comment by bummytime** [comment]
@SergioEstevao a couple of general things:

1) We should probably update dialogs  so it says "videos" or "media" (cc @meremagee):
napkin 04-08-15 1 31 26 pm

2) Videos are not being displayed when creating new posts and editing existing posts. Looks like they are being saved to the server as a `span`: 
```html
<span id="video_container_1A9706BF-7924-400A-8F7F-E45F8ACD0BAC" class="video_container"></span>
```
3) At times, noticed some wacky behavior around images when inserting newlines and/or deleting text/videos with backspace. For example the text would flow under the placeholder/poster image. I think this is related to the 2nd point above.

4) When deleting a video by backspacing over it, the nav bar "uploading" progress bar still displays: https://cloudup.com/cBmDmgY8RAD

**Comment by SergioEstevao** [comment]
@bummytime 

1) Is fixed by this change:https://github.com/wordpress-mobile/WordPress-iOS/commit/b248955df68bd2b16eda317b8e998ec5fd8b2943
2) Is fixed by this change:https://github.com/wordpress-mobile/WordPress-Editor-iOS/pull/626
3) I think it's fixed by the fix in 2 but you may want to test it again
4) This is something that also happens with images they are just quicker to finish the upload. I had a look but there is no easy way to find out that the video was deleted from the post. The only I see it happen is by constantly checking the post content to see if the video tag was deleted and stop the upload for that image. Maybe this should be a new improvement issue?

**Comment by bummytime** [comment]
Looking better @SergioEstevao!

> This is something that also happens with images they are just quicker to finish the upload. I had a look but there is no easy way to find out that the video was deleted from the post. The only I see it happen is by constantly checking the post content to see if the video tag was deleted and stop the upload for that image. Maybe this should be a new improvement issue?

I can open a new issue for this.

Also, regarding deleting videos. I have another vid for you: https://cloudup.com/cjyvInV7SZi Seems like the video will not be removed until the surrounding `&nbsp;`'s are deleted. Also, if the `&nbsp;` in front of a video is removed, the video cannot be deleted (unless the user is in HTML mode).

**Comment by SergioEstevao** [comment]
> Also, regarding deleting videos. I have another vid for you: https://cloudup.com/cjyvInV7SZi Seems like the video will not be removed until the surrounding &nbsp;'s are deleted. Also, if the &nbsp; in front of a video is removed, the video cannot be deleted (unless the user is in HTML mode).

I've being trying to sort this out but it looks something that is part of the edit mode of the webview. If you add the video between text it works ok. The issues only show up if you have a empty post or add a video to the end. 
I've been trying different characters to wrap around the video but they all end up with the same result.

**Comment by bummytime** [comment]
@meremagee @thianhlu  Take a look at @SergioEstevao's [comment above](https://github.com/wordpress-mobile/WordPress-iOS/pull/3503#issuecomment-92150584) and watch the video. Is this a showstopper?

**Comment by bummytime** [comment]
@SergioEstevao Just an update....I spoke with @meremagee offline and we both agreed that this delete issue can be ignored for this PR. When we finally merge, lets create another issue for it.

**Comment by bummytime** [comment]
@SergioEstevao after https://github.com/wordpress-mobile/WordPress-Editor-iOS/pull/626 is merged and the editor hash is updated here, I say :shipit: ...let's get this thing into stabilization! 

Nice job one this!

**Pull request merged**